tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -124,11 +132,34 @@ _ when x.ToBoolean() is { } boolean => boolean,
                          .AsString();
 
             var otelServiceName = config.WithKeys(ConfigurationKeys.OpenTelemetry.ServiceName).AsStringResult();
-            ServiceName = config
+            var serviceName = config
                                  .WithKeys(ConfigurationKeys.ServiceName, "DD_SERVICE_NAME")
                                  .AsStringResult()
                                  .OverrideWith(in otelServiceName, ErrorLog);
 
+            var isUserProvidedTestServiceTag = true;
+            if (isRunningInCiVisibility)
+            {
+                // Set the service name if not set
+                var ciVisServiceName = serviceName;
+                if (string.IsNullOrEmpty(serviceName))
+                {
+                    // Extract repository name from the git url and use it as a default service name.
+                    ciVisServiceName = CIVisibility.GetServiceNameFromRepository(CIEnvironmentValues.Instance.Repository);
+                    isUserProvidedTestServiceTag = false;
+                }
+
+                // Normalize the service name
+                ciVisServiceName = NormalizerTraceProcessor.NormalizeService(ciVisServiceName);
```

Comment:
Maybe we should move the debug log somewhere around there

Reply:
IMO not required. It gets logged by the tracer manager and in telemetry anyway 🤷‍♂️